spring-integration-gcp/src/main/java/org/springframework/integration/gcp/outbound/PubSubMessageHandler.java

```diff
@@ -53,12 +57,44 @@ public PubSubMessageHandler(PubSubOperations pubSubTemplate, String topic) {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) {
-		String payload = (String) this.messageConverter.fromMessage(message, String.class);
+		Object payload = message.getPayload();
+
+		if (payload instanceof PubsubMessage) {
+			this.pubSubTemplate.publish(this.topic, (PubsubMessage) payload);
+			return;
+		}
+
+		ByteString pubsubPayload;
+
+		if (payload instanceof String) {
+			pubsubPayload =
+					ByteString.copyFrom(
+							(String) this.messageConverter.fromMessage(message, String.class),
+							Charset.defaultCharset());
+		}
+		else if (payload instanceof byte[]) {
+			pubsubPayload = ByteString.copyFrom((byte[]) payload);
+		}
+		else if (payload instanceof ByteString) {
+			pubsubPayload = (ByteString) payload;
+		}
+		else {
+			throw new PubSubException("Payload type "
+					+ payload.getClass().getName() + " is unsupported.");
```

Comment:
Would be great to say *what* is supported.
By the way why can't we try to convert from any `Object` to `String` using mentioned `messageConverter`.
Isn't that its purpose and general idea of this PR?
Thanks

Reply:
BTW, that is not good exception here from the `MessageHandler`. There is nothing about PubSub yet.
A `MessageHandlingException` is the right candidate.

Reply:
Ended up getting rid of the exception since we can transform any byte array into a string..